inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/NodeProducer.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.pojo.sort.node;
+
+import org.apache.inlong.manager.common.consts.SourceType;
+import org.apache.inlong.manager.pojo.sort.node.extract.HudiFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.KafkaFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.MongoFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.MysqlBinlogFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.OracleFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.PostgreSqlFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.PulsarFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.RedisFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.SqlServerFactory;
+import org.apache.inlong.manager.pojo.sort.node.extract.TubeMqFactory;
+import org.apache.inlong.manager.pojo.source.StreamSource;
+import org.apache.inlong.manager.pojo.stream.StreamNode;
+import org.apache.inlong.sort.protocol.node.ExtractNode;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The ExtractNode producer.
+ */
+public class NodeProducer {
```

Comment:
Why not just call it `NodeFactory`, or `NodeProvider`? Producer more likes a concept in MQ.

Reply:
> Why not just call it `NodeFactory`, or `NodeProvider`? Producer more likes a concept in MQ.


It has been modified